File: LeetCode/C/145-tree-二叉树的后序遍历.cpp
/*
 * @lc app=leetcode.cn id=145 lang=cpp
 *
 * [145] 二叉树的后序遍历
 */
#include "main.h"
// @lc code=start
class Solution {
  public:
    vector<int> postorderTraversal(TreeNode *root) {
        vector<int> res;

        function<void(TreeNode *)> post = [&](TreeNode *node) {
            if (node == nullptr)
                return;
            post(node->left);
            post(node->right);
            res.push_back(node->val);
        };

        post(root);
        return res;
    }
};
// @lc code=end

File: LeetCode/Go/145-tree-二叉树的后序遍历.go
/*
 * @lc app=leetcode.cn id=145 lang=golang
 *
 * [145] 二叉树的后序遍历
 */
// package leetcode

// @lc code=start
func postorderTraversal(root *TreeNode) []int {
	var res = make([]int, 0)

	var post func(*TreeNode)
	post = func(root *TreeNode) {
		if root == nil {
			return
		}
		post(root.Left)
		post(root.Right)
		res = append(res, root.Val)
	}

	return res
}

// @lc code=end
